guardrails/run/stream_runner.py

```diff
@@ -158,19 +158,27 @@ def step(
             chunk_text = self.get_chunk_text(chunk, api)
             fragment += chunk_text
 
-            # 2. Parse the fragment
-            parsed_fragment, move_to_next = self.parse(
-                index, fragment, output_schema, verified
+            # 2. Parse the chunk
+            # I assume we have to parse the chunk before validating it...
+            parsed_chunk, move_to_next = self.parse(
+                index, chunk, output_schema, verified
             )
             if move_to_next:
                 # Continue to next chunk
                 continue
 
             # 3. Run output validation
+            # If validator chunk size is smaller than LLM chunk size:
```

Comment:
We do not need to do this here. Instead, just pass the whole chunk to each validator. The validator will internally chunk, validate over all validate-able chunks (as determined by the validator), and return a list of validation results.

Reply:
Also important to note: I don't think we should do this validation loop DIRECTLY over here. Instead, we should pass downstream to schema and have this validation logic there, because that's where validator objects are statefully stored